pygeos/set_operations.py

```diff
@@ -16,45 +16,87 @@
 ]
 
 @multithreading_enabled
-def difference(a, b, **kwargs):
+def difference(a, b, grid_size=None, **kwargs):
     """Returns the part of geometry A that does not intersect with geometry B.
 
+    If grid_size is nonzero, input coordinates will be snapped to a precision grid of that
+    size and resulting coordinates will be snapped to that same grid.  If 0, this
+    operation will use double precision coordinates.
+
     Parameters
     ----------
     a : Geometry or array_like
     b : Geometry or array_like
+    grid_size : float, optional (default: None).
+        precision grid size; requires GEOS >= 3.9.0.
```

Comment:
```suggestion
    grid_size : float, optional (default: None)
        Precision grid size; requires GEOS >= 3.9.0.
```

(some formatting nitpicks)